libstuff/libstuff.cpp

```diff
@@ -2522,15 +2505,98 @@ int SQuery(sqlite3* db, const char* e, const string& sql, SQResult& result, int6
+            size_t beforePrepare = 0;
+            if (isSyncThread) {
+                beforePrepare = STimeNow();
+            }
+            error = sqlite3_prepare_v2(db, statementRemainder, strlen(statementRemainder), &preparedStatement, &statementRemainder);
+            if (isSyncThread) {
+                prepareTimeUS += STimeNow() - beforePrepare;
+            }
+             if (error) {
+                // This will just drop through to the general error handling below.
+                sqlite3_finalize(preparedStatement);
+                break;
+            } else if (!preparedStatement) {
+                // If we get a null statement (from parsing a blank string) we can skip, this isn't an error.
+                error = SQLITE_OK;
+                break;
+            }
+            int numColumns = sqlite3_column_count(preparedStatement);
+            result.headers.resize(numColumns);
+
+            while (true) {
+                size_t beforeStep = 0;
+                if (isSyncThread) {
+                    beforeStep = STimeNow();
+                }
+                numSteps++;
+                error = sqlite3_step(preparedStatement);
+                if (isSyncThread) {
+                    size_t stepTime = STimeNow() - beforeStep;
+                    if (stepTime > longestStepTimeUS) {
+                        longestStepTimeUS += stepTime;
+                    }
+                    stepTimeUS += stepTime;
+                }
+
+                for (int i = 0; i < numColumns; i++) {
+                    result.headers[i] = sqlite3_column_name(preparedStatement, i);
+                }
+
+                if (error == SQLITE_ROW) {
+                    result.rows.emplace_back(vector<string>(numColumns));
+                    for (int i = 0; i < numColumns; i++) {
+                        int colType = sqlite3_column_type(preparedStatement, i);
+                        switch (colType) {
+                            case SQLITE_INTEGER:
+                                result.rows.back()[i] = to_string(sqlite3_column_int64(preparedStatement, i));
+                                break;
+                            case SQLITE_FLOAT:
+                                result.rows.back()[i] = to_string(sqlite3_column_double(preparedStatement, i));
+                                break;
+                            case SQLITE_TEXT:
+                                result.rows.back()[i] = reinterpret_cast<const char*>(sqlite3_column_text(preparedStatement, i));
+                                break;
+                            case SQLITE_BLOB:
+                                result.rows.back()[i] = string(static_cast<const char*>(sqlite3_column_blob(preparedStatement, i)), sqlite3_column_bytes(preparedStatement, i));
+                                break;
+                            case SQLITE_NULL:
+                                // null string.
+                                break;
```

Comment:
Why is everything being casted to a string? Or maybe a better question is how can I verify that this whole block is correct? I looked through the implementation of `sqlite3_exec` in libstuff, I could look a little more carefully if that's the answer

Reply:
Because it always has before. SQResult holds a bunch of strings. See:
https://github.com/Expensify/Bedrock/pull/1399/files#diff-d0f3e9340d9d47b780f0bf98f9eb8e1edb6f041652946c35fcae859e2ea82ebcL2512